packages/browser/src/plugins/remote-loader/index.ts

```diff
@@ -100,7 +190,26 @@ export async function remoteLoader(
 
           validate(plugins)
 
-          allPlugins.push(...plugins)
+          const routing = routingRules.filter(
+            (rule) => rule.destinationName === remotePlugin.creationName
+          )
+
+          plugins.forEach((plugin) => {
+            const wrapper = new ActionDestination(
+              remotePlugin.creationName,
+              plugin
+            )
+
+            if (
+              routing.length &&
+              routingMiddleware &&
+              plugin.type === 'destination'
```

Comment:
Might be worth adding a comment that we only want to apply destination filters to `destination` actions so that we don't cause issues for hybrid action destinations (where we simply augment the event and send it to the cloud rather than directly to the destination)

I see future selves forgetting why we limited this :smile: